src/uct/tcp/sockcm/sockcm_ep.c

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2017-2019.  ALL RIGHTS RESERVED.
+ * Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "sockcm_ep.h"
+
+#define UCT_SOCKCM_CB_FLAGS_CHECK(_flags) \
+    do { \
+        UCT_CB_FLAGS_CHECK(_flags); \
+        if (!((_flags) & UCT_CB_FLAG_ASYNC)) { \
+            return UCS_ERR_UNSUPPORTED; \
+        } \
+    } while (0)
+
+static UCS_CLASS_INIT_FUNC(uct_sockcm_ep_t, const uct_ep_params_t *params)
+{
+    uct_sockcm_iface_t *iface = ucs_derived_of(params->iface,
+                                               uct_sockcm_iface_t);
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &iface->super);
+    return UCS_OK;
+}
+
+static UCS_CLASS_CLEANUP_FUNC(uct_sockcm_ep_t)
+{
+}
+
+UCS_CLASS_DEFINE(uct_sockcm_ep_t, uct_base_ep_t)
+UCS_CLASS_DEFINE_NEW_FUNC(uct_sockcm_ep_t, uct_ep_t, const uct_ep_params_t *);
+UCS_CLASS_DEFINE_DELETE_FUNC(uct_sockcm_ep_t, uct_ep_t);
+
+void uct_sockcm_ep_set_failed(uct_iface_t *iface, uct_ep_h ep, ucs_status_t status)
+{
+    ucs_trace("uct_sockcm_ep_set_failed not implemented");
+}
+
+void uct_sockcm_ep_invoke_completions(uct_sockcm_ep_t *ep, ucs_status_t status)
+{
+    uct_sockcm_ep_op_t *op;
+
+    ucs_assert(pthread_mutex_trylock(&ep->ops_mutex) == EBUSY);
+
+    ucs_queue_for_each_extract(op, &ep->ops, queue_elem, 1) {
+        pthread_mutex_unlock(&ep->ops_mutex);
```

Comment:
can we use `ucs_spinlock_t` instead of `mutex`?
`spinlock` has `trylock` method as well

Reply:
i don't think it's important in this context